### PR TITLE
1622: Added samoa confirmed cases overlay

### PR DESCRIPTION
**Pull request by chris-bes**
### Changes:
- Added samoa confirmed cases overlay

### CODE REVIEW

**Comment by edmofro** [comment]
@chris-bes could you update this from dev? I'm going to review assuming that only https://github.com/beyondessential/tupaia/pull/2172/commits/b77cb989d26bb2299023cd1ce3a38c0fb03b6ad4 is relevant

File: packages/database/src/migrations/20210219055420-SamoaHeatmapHomeVillageOfConfirmedCases-modifies-data.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      "${key}" = '${value}';
+  `);
+}
+const indicator = {
+  id: generateId(),
+  code: 'COVID_SAMOA_Confirmed_Cases',
+  builder: 'eventCheckConditions',
+  config: {
+    formula:
+      "WS_COVID19_Clinsurv_137 == 1 or equalText(WS_COVID19_Clinsurv_75, '1. Positive') or equalText(WS_COVID19_Clinsurv_79, '1. Positive') or equalText(WS_COVID19_Clinsurv_83, '1. IgG positive only') or equalText(WS_COVID19_Clinsurv_83, '2. IgM positive only') or equalText(WS_COVID19_Clinsurv_83, '3. IgM / IgG positive') or equalText(WS_COVID19_Clinsurv_87, '1. Positive')",
```
Comment (edmofro) [inline packages/database/src/migrations/20210219055420-SamoaHeatmapHomeVillageOfConfirmedCases-modifies-data.js:34]: Can formulae contain new lines? I reckon we should have a policy of pulling these out into variables and formatting them for easier reading, it's a little tough to get through when all in one line

File: packages/database/src/migrations/20210219055420-SamoaHeatmapHomeVillageOfConfirmedCases-modifies-data.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    programCode: 'SC1CS',
+    defaultValues: {
+      WS_COVID19_Clinsurv_137: 0,
+      WS_COVID19_Clinsurv_75: 'undefined',
```
Comment (edmofro) [inline packages/database/src/migrations/20210219055420-SamoaHeatmapHomeVillageOfConfirmedCases-modifies-data.js:38]: What is the default default value? Do we need to specify the `undefined` ones?

File: packages/database/src/migrations/20210219055420-SamoaHeatmapHomeVillageOfConfirmedCases-modifies-data.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+async function deleteItemByOneCondition(db, table, condition) {
```
Comment (edmofro) [inline packages/database/src/migrations/20210219055420-SamoaHeatmapHomeVillageOfConfirmedCases-modifies-data.js:19]: I'd be interested to have a discussion about where to define things that will be used in only the `down` migration. Personally I'd be keen to push them to just above the `down`, but I know we have a bit of a convention of this pattern of defining everything here in the migration file. Perhaps you could raise it at dev chat?

File: packages/database/src/migrations/20210219055420-SamoaHeatmapHomeVillageOfConfirmedCases-modifies-data.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      null: true,
+    },
+  },
+  countryCodes: '{WS}',
+  projectCodes: '{covid_samoa}',
+};
+exports.up = async function (db) {
+  await insertObject(db, 'mapOverlay', mapOverlay);
+
+  const mapOverLayGroupId = await codeToId(db, 'map_overlay_group', 'COVID19_Samoa');
```
Comment (edmofro) [inline packages/database/src/migrations/20210219055420-SamoaHeatmapHomeVillageOfConfirmedCases-modifies-data.js:71]: Extremely nitpicky, but no capital on the 'L' of overlay

**Review by edmofro — APPROVED** [review]
Only some suggestions, but approved (again, assuming it's just the one commit that's in the diff once conflicts are resolved)

File: packages/database/src/migrations/20210219055420-SamoaHeatmapHomeVillageOfConfirmedCases-modifies-data.js (hunk 2 above)
Comment (chris-bes) [reply packages/database/src/migrations/20210219055420-SamoaHeatmapHomeVillageOfConfirmedCases-modifies-data.js:38]: Yeah we definitely don't need it now I think about it

**Comment by chris-bes** [comment]
This branch has a lot of conflicts with dev so I thought it'd be easier if I just make a new branch/PR (https://github.com/beyondessential/tupaia/pull/2199) and close this one. I've addressed the review comments in this commit (https://github.com/beyondessential/tupaia/pull/2199/commits/060d2d40e404f1030f4c6e9641b5a97ab8b1cc4c)

**Pull request closed without merging**